dockers/docker-sonic-restapi/cert_monitor.py

```diff
@@ -0,0 +1,111 @@
+    while True:
+        if os.path.isfile(path_to_cert_tracker):
+            # load last read 'last modified' times
+            cert_tracker = json.load(open(path_to_cert_tracker))
+        else:
+            cert_tracker = {}
+        restart = False
+        for certificate in certificates:
+            if os.path.isfile(certificate):
+                # Get 'last modified' time of the symbolic link of the cert
+                m_time = os.lstat(certificate).st_mtime
+                if certificate in cert_tracker:
+                    if cert_tracker[certificate] != m_time:
+                        logger.log_info("monitor_certificates: Certificate has rolled. Restart required")
+                        restart = True
+                cert_tracker[certificate] = m_time
+            else:
+                logger.log_error("monitor_certificates: "+certificate+" not found!")
+        # Save 'last modifed' times
+        json.dump(cert_tracker, open(path_to_cert_tracker, "w"))
+
+        if restart:
+            pid = get_pid(RESTAPI_SERVER_PROCESS)
+            if pid:
+                # Send SIGQUIT to go-server-server
+                cmd = "kill -3 "+pid
+                response = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                stdout, stderr = response.communicate()
+                if (response.returncode != 0):
+                    logger.log_error(stderr, True)
+                else:
+                    logger.log_info("monitor_certificates: SIGQUIT sent to go-server-server")
+                    while get_pid(RESTAPI_SERVER_PROCESS) != None:
+                        time.sleep(0.1)
+                    logger.log_info("monitor_certificates: go-server-server has shutdown")
+            else:
+                logger.log_warn("monitor_certificates: RESTAPI Server (go-server-server) is not running!")
+            
+            # Safe to restart restapi server now
+            cmd = "supervisorctl restart restapi"
+            response = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            stdout, stderr = response.communicate()
+            if (response.returncode != 0):
+                logger.log_error(stderr, True)
+                logger.log_info("monitor_certificates: RESTAPI Server restart failed")
+            else:
+                logger.log_info("monitor_certificates: RESTAPI Server has been restarted")
+
+        time.sleep(POLLING_FREQUENCY)
+
+
+if __name__ == '__main__':
+    config_db = swsscommon.DBConnector("CONFIG_DB", REDIS_TIMEOUT_MS, True)
+    restapi_table = swsscommon.Table(config_db, "RESTAPI")
+    (status, tuples) = restapi_table.get("certs")
+    restapi_certs = dict(tuples)
+
+    certificates = []
+    certificates.append(restapi_certs['client_ca_crt'])
+    certificates.append(restapi_certs['server_crt'])
+    certificates.append(restapi_certs['server_key'])
+
+    # wait for go-server-server to start
+    while True:
```

Comment:
Looks like this loop is not required. isn't monitor function already handling this?

Reply:
No, this is specifically for waiting for go-server-server process to begin.

Reply:
this can be handled by dependent start introduced in earlier pr. you can use that, and make your process to require go-server-server to start first.